Frameworks/Foundation/NSProgress.mm

```diff
@@ -46,16 +46,15 @@
     bool childCreated;
 };
 
-// Stack of NSProgress objects that have becomeCurrent
-thread_local static std::shared_ptr<std::stack<CurrentProgress>> s_currentProgressStack;
-
 // Returns the stack of NSProgress objects that have becomeCurrent on the current thread, initializing it if necessary
-static decltype(s_currentProgressStack) & _getProgressStackForCurrentThread() {
-    if (!s_currentProgressStack) {
-        s_currentProgressStack = std::make_shared<std::stack<CurrentProgress>>();
+static std::shared_ptr<std::stack<CurrentProgress>> & _getProgressStackForCurrentThread() {
+    thread_local static std::shared_ptr<std::stack<CurrentProgress>> tlsCurrentProgressStack;
+
+    if (!tlsCurrentProgressStack) {
+        tlsCurrentProgressStack = std::make_shared<std::stack<CurrentProgress>>();
```

Comment:
@ehren we do that in a lot of places where the type is expressed fully on the righthand side :)